pkg/controller/master/upgrade/upgrade_repo.go

```diff
@@ -556,3 +383,115 @@ func (r *Repo) getImagesDiffList() ([]string, error) {
 
 	return diffList, nil
 }
+
+func (r *Repo) CreateDeployment(vmImage *harvesterv1.VirtualMachineImage) (*appsv1.Deployment, error) {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.GetDeploymentName(),
+			Namespace: upgradeNamespace,
+			Labels: map[string]string{
+				harvesterUpgradeLabel:          r.upgrade.Name,
+				harvesterUpgradeComponentLabel: upgradeComponentRepo,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				upgradeReference(r.upgrade),
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.To(int32(3)),
```

Comment:
is `3` hard coded? could we convert it to a param of the CreateDeployment and allow the customization,

e.g. for single-node upgrade, spawn a single replica deployment

Reply:
Or maybe set the `management.cattle.io/scale-available` annotation to `3` if we don't mind relying on Rancher to handle the auto-scaling.